test/testcafe/spec/SessionStorage_spec.js

```diff
@@ -261,3 +263,59 @@ test.requestHooks(credentialSSONotExistLogger, credentialSSONotExistMock)('shall
   )).eql(1);
   await t.expect(getStateHandleFromSessionStorage()).eql(null);
 });
+
+test.requestHooks(identifyChallengeMock)('shall back to sign-in and authenticate succesfully', async t => {
+  const identityPage = new IdentityPageObject(t);
+  const challengeEmailPageObject = new ChallengeEmailPageObject(t);
+  let pageTitle;
+
+  // Add mocks for interaction code flow
+  const challengeSuccessMock = RequestMock()
+    .onRequestTo('http://localhost:3000/idp/idx/challenge/answer')
+    .respond(xhrSuccessWithInteractionCode)
+    .onRequestTo('http://localhost:3000/oauth2/default/v1/token')
+    .respond(xhrSuccessTokens);
+  await t.addRequestHooks(challengeSuccessMock);
+
+  // Setup widget with interaction code flow
+  const optionsForInteractionCodeFlow = {
+    clientId: 'fake',
+    useInteractionCodeFlow: true,
+    authParams: {
+      ignoreSignature: true,
+      pkce: true,
+    },
+    stateToken: undefined
+  };
+  await identityPage.navigateToPage({ render: false });
+  await identityPage.mockCrypto();
+  await t.setNativeDialogHandler(() => true);
+  await renderWidget(optionsForInteractionCodeFlow);
+
+  // Identify page
+  await identityPage.fillIdentifierField('foo@test.com');
+  await identityPage.clickNextButton();
+
+  // Email challenge page - click 'Back to sign-in'
+  pageTitle = challengeEmailPageObject.form.getTitle();
+  await t.expect(pageTitle).eql('Verify with your email');
+  await challengeEmailPageObject.clickSignOutLink();
+
+  // Go back to Identify page
+  pageTitle = identityPage.form.getTitle();
+  await t.expect(pageTitle).eql('Sign In');
+  await identityPage.fillIdentifierField('foo@test.com');
+  await identityPage.clickNextButton();
+
+  // Email challenge page - verify
+  pageTitle = challengeEmailPageObject.form.getTitle();
+  await t.expect(pageTitle).eql('Verify with your email');
+  await challengeEmailPageObject.verifyFactor('credentials.passcode', '1234');
+  await challengeEmailPageObject.clickNextButton();
+  
+  // Check success
+  const history = await t.getNativeDialogHistory();
+  await t
+    .expect(history.length).eql(1)
+    .expect(history[0].text).eql('SUCCESS: OIDC with single responseType. Check Console');
```

Comment:
Instead of checking for a prompt, we can get more value from parsing the console logs to validate the correct [response](https://github.com/okta/okta-signin-widget/blob/17682c785c97f00a6e53911c455c61841b2798ae/playground/main.js#L62).

Reply:
It would be great, but TestCafe can't return object used in `console.log`, it returns string `[object Object]`
There is open issue for this:
https://github.com/DevExpress/testcafe/issues/5425

Reply:
We should still be able to compare object equality though, as we're doing using the `checkConsoleMessages` helper. Here is an example where we're [checking for context](https://github.com/okta/okta-signin-widget/blob/d0f55fee688fe6c79620cf7e37e7a53b236311e4/test/testcafe/spec/IdentifyRegistrationWithCaptcha_spec.js#L49-L52).

Reply:
Changed.
See https://github.com/okta/okta-signin-widget/pull/2090/files#diff-d5619ca161ed5c74752d62d68598cc9038e1a588eee3755c9f3a103ed3ad43e0R317